common/utils.js

```diff
@@ -857,11 +857,56 @@
             return $window.location.origin + "/id/" + currCatalog + "/" + tuple.data.RID + (version ? version : "");
         }
 
+        // if '?' is used instead of '#' (?catalog/schema:table), return in the proper form (#catalog/schema:table)
+        function getHash(location) {
+            var hash = location.hash
+
+            // allow ? to be used in place of #
+            if ((hash == '' || hash == undefined) && location.href.indexOf("?") !== -1) {
+                hash = "#" + location.href.substring(location.href.indexOf("?") + 1);
+            }
+
+            return hash;
+        }
+        /**
+         * @param {String} hash - window.location.hash string
+         */
+        function stripSortAndQueryParams(hash) {
+            // '@' appears first, search for that before searching for '?'
+            if (hash.indexOf('@') > -1) {
+                return hash.split('@')[0];
+            } else {
+                return hash.split('?')[0];
+            }
+        }
+
         function removeParentContext(url) {
             url = url.substring(0, url.lastIndexOf("?"));
             $window.history.replaceState('', '', url);
         }
 
+        /**
+         * Given a location href and key, return the query param value that matches that key
+         * @param {String} url - the full url for the current page
+         * @param {String} key - the key of the query parameter you want the value of
+         * @returns {String} the value of that key or null, if that key doesn't exist as a query parameter
+         *
+         * Note: This won't handle the case where the url might be like this:
+         * '?catalog/schema:table/limit=20' where limit is a column name
+         */
+        function getQueryParam(url, key) {
```

Comment:
Did you try this with urls that start with `?`? I feel like this might not behave as you expect since you're splitting them by `&` (and if you have filters with & this will not behave properly). 

Since the function is accepting a "key", you don't need to extract all the query parameters, you just need to look for the `key + "="` in the substring that starts from the `url.lastIndexOf("?")`. And then return anything that comes after the `=` and before `&` if exists.

Reply:
I agree, it's probably better to just look for the key being present and strip the contents after the `=`. 

About your first point, do you mean if the function were called like this `getQueryParam("www.domain.org/chaise/record/?catalog/schema:table/limit=20", "limit")`?